### ISSUE TITLE
zdb.c received SIGSEGV at zdb.c:491

### ISSUE THREAD

**Issue by RichardSharpe** [zfsonlinux/zfs#4245]
Hi,

Has anyone seen this? 

I was using zdb -dddd some-zpool on one of our pools that was causing problems and it crashed at the above location.

Much more info available on request.

**Comment by RichardSharpe** [comment]
Here is the last useful output:
```
    Object  lvl   iblk   dblk  dsize  lsize   %full  type
       379    2    16K     4K  90.0K  40.0K  100.00  SPA space map
                                        320   bonus  SPA space map header
        dnode flags: USED_BYTES
        dnode maxblkid: 9

    Object  lvl   iblk   dblk  dsize  lsize   %full  type
       380    1    16K   128K      0   128K    0.00  bpobj subobj
        dnode flags: USED_BYTES
        dnode maxblkid: 0
        0
        0
        0
        0
        0
        0
```
After that it is lines and lines of 0 until it crashed.

**Comment by dweeezil** [comment]
Should be fixed by 2986b3f.  That patch isn't in any tag yet.

**Comment by RichardSharpe** [comment]
Thanks

**Comment by loli10K** [comment]
The fix is now in the 0.6.5 branch https://github.com/zfsonlinux/zfs/commit/34a3a7c660b014754b6b97dbff6f6ef94626255a, i think we can close this.
